### PR TITLE
cleanup(sidekick/swift): library vs. package names

### PR DESCRIPTION
**Pull request by coryan**
Use different annotations for the package name (e.g. `google-cloud-workflows-v1`) vs. the library name (e.g. `GoogleCloudWorkflowsV1`). In a future PR we will change the default library name and introduce options to override it.

Towards #6229

### CODE REVIEW

File: internal/sidekick/swift/library_name.go [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+import (
+	"strings"
+
+	"github.com/googleapis/librarian/internal/config"
+	"github.com/googleapis/librarian/internal/sidekick/api"
+	"github.com/iancoleman/strcase"
+)
+
+// LibraryName returns the Swift library (and module) name for the API.
+func LibraryName(api *api.API, swiftCfg *config.SwiftPackage) string {
```
Comment (nolanleastin) [inline internal/sidekick/swift/library_name.go]: nit: swiftCfg is not used

File: internal/sidekick/swift/library_name.go [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+	// a better default.
+	parts := strings.Split(api.PackageName, ".")
+	for i, p := range parts {
+		parts[i] = strcase.ToCamel(p)
+	}
+	result := strings.Join(parts, "")
+	if strings.HasPrefix(result, "Google") {
+		return result
+	}
+	return "Google" + result
```
Comment (nolanleastin) [inline internal/sidekick/swift/library_name.go]: If PackageName is empty, we will return `Google`. Not sure that's what you want.

File: internal/sidekick/swift/codec.go [hunk 3]
```diff
@@ -54,13 +54,22 @@ type codec struct {
...
+	// LibraryName is the name of the Swift library (e.g. "GoogleCloudSecretManagerV1").
+	//
+	// Note that GAPIC packages contain a single product (the library), which
+	// contains a single target and module with the same names as the library.
+	LibraryName string
+
+	// The name of the Swift package (e.g. "google-cloud-secretmanager-v1").
+	//
+	// Note that GAPIC packages contain a single product (the library), which
+	// contains a single target and module with the same names as the library.
```
Comment (nolanleastin) [inline internal/sidekick/swift/codec.go]: Is this duplicated comment necessary?

File: internal/sidekick/swift/library_name_test.go [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+		{
+			name:  "api",
+			input: "google.api",
+			want:  "GoogleApi",
+		},
+		{
+			name:  "grafeas v1",
+			input: "grafeas.v1",
+			want:  "GoogleGrafeasV1",
+		},
```
Comment (nolanleastin) [inline internal/sidekick/swift/library_name_test.go:53]: nit: some corner cases: 

input "google" results in "Google"
input: "" results in "Google" 

if this is what we want, add tests for them.

**Review by nolanleastin — APPROVED** [review]
should generate_package_swift_test.go test this too?

File: internal/sidekick/swift/codec.go (hunk 3 above)
Comment (coryan) [reply internal/sidekick/swift/codec.go]: Removed.

File: internal/sidekick/swift/library_name.go (hunk 1 above)
Comment (coryan) [reply internal/sidekick/swift/library_name.go]: Removed, I will add it back in a future PR when we start using it...

File: internal/sidekick/swift/library_name.go (hunk 2 above)
Comment (coryan) [reply internal/sidekick/swift/library_name.go]: I am pretty sure that should be an error, so I made it one.  Which (sadly) propagated to many tests that did not initialize the PackageName correctly.

PS: the new `api.NewTest*()` helpers take care of initializing PackageName  automatically, we should use them everywhere.

File: internal/sidekick/swift/library_name_test.go (hunk 4 above)
Comment (coryan) [reply internal/sidekick/swift/library_name_test.go:53]: Fixed the code to handle the empty string and included a test for the Google case. Not that we care what the answer is, that (and many other) package names are never going to show up for realsies.

**Comment by coryan** [comment]
> should generate_package_swift_test.go test this too?

Done.
